src/Detector/Node/NodeDetector.cs

```diff
@@ -151,5 +149,57 @@ private dynamic ReadJsonObjectFromFile(ISourceRepo sourceRepo, string fileName)
             var jsonContent = sourceRepo.ReadFile(fileName);
             return JsonConvert.DeserializeObject(jsonContent);
         }
+
+        private IEnumerable<FrameworkInfo> DetectFrameworkInfos(DetectorContext context)
+        {
+            var detectedFrameworkResult = new List<FrameworkInfo>();
+            var packageJson = GetPackageJsonObject(context.SourceRepo, _logger);
+            if (packageJson?.devDependencies != null)
+            {
+                var devDependencyObject = packageJson?.devDependencies;
+                foreach (var KeyWordToName in NodeConstants.DevDependencyFrameworkKeyWordToName)
+                {
+                    var keyword = KeyWordToName.Key;
+                    if (devDependencyObject[keyword] != null)
```

Comment:
Would this check throw if the key is not present? Also can you check if this does a case-INsensitive way of lookup?

Reply:
It will throw exception if no key found, added the catch exception here. And I checked it's case insensitive by default.